pkg/netlight/resource/mycelium.go

```diff
@@ -178,17 +182,31 @@ func setupMycelium(netNS ns.NetNS, mycelium string, seed []byte) error {
 		return err
 	}
 
-	name := filepath.Base(netNS.Path())
-	if err := os.WriteFile(filepath.Join(MyceliumSeedDir, name), seed, 0444); err != nil {
-		return fmt.Errorf("failed to create seed file '%s': %w", name, err)
-	}
+	if netNS != nil {
+
+		name := filepath.Base(netNS.Path())
+		if err := os.WriteFile(filepath.Join(MyceliumSeedDir, name), seed, 0444); err != nil {
+			return fmt.Errorf("failed to create seed file '%s': %w", name, err)
+		}
 
-	return ensureMyceliumService(zinit.Default(), name, list)
+		return ensureMyceliumService(zinit.Default(), &name, list)
+	} else {
```

Comment:
you already return here, no need for else

Reply:
how?

Reply:
the code in the else clause can be in the function body itself and we can remove the else bs

Reply:
ahha ok, I thought u meant remove the else code all together